### DISCUSSION TITLE
How to detect a table without many borders

### DISCUSSION THREAD

**Discussion by NoahPerkovich** [pymupdf/PyMuPDF-Utilities#130] [Q&A]
Hi all, I am wondering if you have advice for detecting a table that looks like this with your page.find_tables() function? 
![image](https://github.com/pymupdf/PyMuPDF-Utilities/assets/38167243/3a87c062-3a81-4f9c-9ee9-c381106d9321)

**Comment by NoahPerkovich** [comment]
Note, this table is in a larger PDF of an academic paper, I am hoping to scrape multiple tables from this document, and from other documents

**Comment by JorjMcKie** [comment]
There is not much hope here. Columns seem to be fairly recognizable, rows definitely are not:
Even a human eye has problems to find a suitable principle that tells it that e.g. this is meant to represent 2 rows and not 9:
![image](https://github.com/pymupdf/PyMuPDF-Utilities/assets/8290722/2de5e468-da44-48c0-bea4-e2dc1f61a492)

You can specify additional, "virtual" lines in `find_tables()` via the `add_lines` parameter. This is a list of pairs of points. These lines will be treated as if they had actually been drawn.
A possible strategy may consist of:
1. Locate the header by searching for words "Country", "Reference", ... having the same bottom coordinate. The top coordinate of these words is the table top. The left coordinate designate the column boundaries. This information can be used to define 2 horizontal lines (above and below the header words) and 5 or more vertical lines, the identified column borders.
2. The union of all points we have identified so far for defining the lines are used to define a clip rectangle for the table.
3. When feeding this into `find_tables(clip=clip, add_lines=line_list)` we will get at least **_some_** table. Probably, the table rows will be wrong though ...
